### DISCUSSION TITLE
"KMS: DRM_IOCTL_MODE_CREATE_DUMB failed: Permission denied" when I try to open some sites

### DISCUSSION THREAD

**Discussion by decembre-dhiver** [elementary/browser#104] [Q&A]
Hi! I need help. Do not know whether I should create a new Issue or not

The problem: when I open some sites I see "Oops! Something went wrong while displaying this page. Please reload or visit a different page to continue." page. And terminal says this:

"KMS: DRM_IOCTL_MODE_CREATE_DUMB failed: Permission denied

(epiphany:2): epiphany-WARNING **: 00:01:04.497: Web process crashed"

It happens every time when I try to open any youtube video or https://www.reddit.com/r/elementaryos/ for example

I can not find any solution to this, so may be someone here would help.

**Comment by mcatanzaro** [comment]
I think this is fixed in WebKitGTK 2.40.1.

**Reply by decembre-dhiver** [reply]
Thank you for the answer!

As far as I understand the current version of the web (flatpak for the AppCenter) is build upon 2.39+, right? So I just have to wait until it will be upgraded to 2.40.1? Or are there any actions that I could take to solve this?

**Reply by mcatanzaro** [reply]
2.39 is an unstable/development version that should not be shipped to users.

If elementary is not shipping 2.40.1, then you should report a bug (here on this project).

**Reply by danirabbit** [reply]
We're shipping with 2.40.0 at the moment as we're building on the GNOME 44 runtime

**Reply by mcatanzaro** [reply]
The GNOME 44 runtime has 2.40.1.

**Comment by decembre-dhiver** [comment]
Found this bug report on [webkitbugzilla](https://bugs.webkit.org/show_bug.cgi?id=254807). It's actually solved, but I still have the problem. So it's possible to use WEBKIT_DISABLE_DMABUF_RENDERER=1 as a workaround. Well... I don't know how to enable it =3. Typing "WEBKIT_DISABLE_DMABUF_RENDERER=1" as a environmental variable in  Flatseal does not change anything =((

**Reply by mcatanzaro** [reply]
That environment variable does nothing unless you have WebKitGTK 2.41.1 or newer, which you don't. The dmabuf renderer is not available in 2.40.

**Reply by decembre-dhiver** [reply]
So yeah... As a casual user I won't be able to figure this out, but I really like the look and feel of Epiphany and would like to see it completely functioning =/

**Reply by csaltos** [reply]
Thank you !! ... This works for me !! :+1::sunglasses:

**Comment by nobane** [comment]
I ran into this issue trying use webkit2gtk on arch linux (with an nvidia card) and I was able to fix it **without** setting `WEBKIT_DISABLE_DMABUF_RENDERER=1`.

Here is how I fixed it:


1. Check if you have modesetting enabled via

        cat /sys/module/nvidia_drm/parameters/modeset

2. If you see printed `N` then you need to enable it with

        echo options nvidia_drm modeset=1 | sudo tee /etc/modprobe.d/nvidia_drm.conf

3. Restart your computer

**Reply by bobo-the-hobo** [reply]
I just had this issue with Bambu studio, 3d printer program. This seemed to fix it. 
Arch / Nvidia 
Thanks!

**Reply by Lucas-Froguel** [reply]
Had this issue while using Mass Effect save editor and this saved (arch+nvidia as well). Thanks!

**Reply by rustbuckett** [reply]
Mine returns 'Y', but I still have the issue. I'm using libwebkit2gtk-4.1-0 2.46.5-1 on Debian 12 with nVidia drivers. `env WEBKIT_DISABLE_DMABUF_RENDERER=1 ./whatever` made it work.

**Reply by ezraclintoc** [reply]
unfortnutaally this didn't work for me, it showed up for a split second then it crashed
